python/lsst/source/injection/consolidate_injected_catalogs.py

```diff
@@ -0,0 +1,283 @@
+        storageClass="ArrowAstropy",
+        minimum=0,
+        multiple=True,
+    )
+    skyMap = Input(
+        doc="Input definition of geometry/bbox and projection/wcs for warped exposures",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+    output_catalog = Output(
+        doc="Per-tract multiband catalog of injected sources.",
+        name="{injected_prefix}deepCoadd_catalog_tract",
+        storageClass="ArrowAstropy",
+        dimensions=("skymap", "tract"),
+    )
+
+
+class ConsolidateInjectedCatalogsConfig(  # type: ignore [call-arg]
+    PipelineTaskConfig, pipelineConnections=ConsolidateInjectedCatalogsConnections
+):
+    """Base configuration for source injection tasks."""
+
+    col_ra = Field[str](
+        doc="Column name for right ascension (in degrees).",
+        default="ra",
+    )
+    col_dec = Field[str](
+        doc="Column name for declination (in degrees).",
+        default="dec",
+    )
+    col_mag = Field[str](
+        doc="Column name for magnitude.",
+        default="mag",
+    )
+    col_source_type = Field[str](
+        doc="Column name for the source type used in the input catalog. Must match one of the surface "
+        "brightness profiles defined by GalSim.",
+        default="source_type",
+    )
+
+
+class ConsolidateInjectedCatalogsTask(PipelineTask):
+    """Class for combining all tables in a collection of input catalogs
+    into one table.
+    """
+
+    _DefaultName = "consolidateInjectedCatalogsTask"
+    ConfigClass = ConsolidateInjectedCatalogsConfig
+
+    def runQuantum(self, butler_quantum_context, input_refs, output_refs):
+        inputs = butler_quantum_context.get(input_refs)
+        catalog_dict, tract = self._get_catalogs(inputs, input_refs)
+        outputs = self.run(catalog_dict, inputs["skyMap"], tract)
+        butler_quantum_context.put(outputs, output_refs)
+
+    def _get_catalogs(
+        self,
+        inputs: dict,
+        input_refs: InputQuantizedConnection,
+    ) -> tuple[dict, int]:
+        """Organize input catalogs into a dictionary with photometry band
+        keys.
+        """
+        catalog_dict: dict[str, list] = {}
+        tracts = set()
+        for ref, catalog in zip(input_refs.input_catalogs, inputs["input_catalogs"]):
+            band = ref.dataId.band.name
+            if band not in catalog_dict:
+                catalog_dict[band] = []
+            # load the patch number to check for patch overlap duplicates later
+            catalog["patch"] = ref.dataId.patch.id
+            catalog_dict[band].append(catalog)
+            tracts.add(ref.dataId.tract.id)
+        # vstack all the catalogs for each band
+        for band, catalog_list in catalog_dict.items():
+            catalog_dict[band] = vstack(catalog_list)
+        if len(tracts) != 1:
+            raise RuntimeError(f"len({tracts=}) != 1")
+        return (catalog_dict, list(tracts)[0])
```

Comment:
This just picks the first tract from the list?

Reply:
There should only be one tract in the list, so it should be taking the only tract